homeassistant/components/sensor/socialblade.py

```diff
@@ -0,0 +1,88 @@
+"""
+Support for Social Blade .
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.socialblade/
+"""
+
+import logging
+from datetime import timedelta
+import voluptuous as vol
+
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.helpers import config_validation as cv
+from homeassistant.const import CONF_FRIENDLY_NAME
+
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['socialbladeclient==0.2']
+
+DEFAULT_NAME = "Social Blade"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_FRIENDLY_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required('channel_id'): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setting the platform in HASS and Channel Information."""
+    social_blade = SocialBladeSensor(config['channel_id'],
+                                     config[CONF_FRIENDLY_NAME])
+    social_blade.update()
+    if social_blade.valid_channel_id:
+        add_devices([social_blade])
```

Comment:
discuss with @fabaff we will keep it as is since update getting called to setup socalblade.valid_channel_id then we check if we need to add to devices()